src/app/models/sap_model.php

```diff
@@ -55,11 +55,19 @@ public function is_admin($email) {
         return boolval($row[0]);
     }
 
-    public function get_missions() {
-        $result = $this->DB->sap->query(
-            'SELECT `id`, `level`, `title`, `description` FROM `sap_missions` ' .
-            'ORDER BY level ASC, id ASC'
-        );
+    public function get_missions($get_unpublished) {
+        if ($get_unpublished) {
+            $result = $this->DB->sap->query(
+                "SELECT `id`, `level`, `title`, `description` FROM `sap_missions`
+                 ORDER BY level ASC, id ASC"
+            );
+        } else {
+            $result = $this->DB->sap->query(
+            "SELECT `id`, `level`, `title`, `description` FROM `sap_missions`
+             WHERE `published` IS TRUE
+             ORDER BY level ASC, id ASC"
+            );
+        }
```

Comment:
I think keep a single query with conditional appending of `WHERE` clause. Updating this query will be a pain.